yt_dlp/extractor/youtube.py

```diff
@@ -627,32 +632,66 @@ def _extract_context(self, ytcfg=None, default_client='web'):
         client_context.update({'hl': self._preferred_lang or 'en', 'timeZone': 'UTC', 'utcOffsetMinutes': 0})
         return context
 
-    _SAPISID = None
+    @staticmethod
+    def _make_sid_authorization(scheme, sid, origin, additional_parts):
+        timestamp = str(round(time.time()))
+
+        hash_parts = []
+        if additional_parts:
+            hash_parts.append(':'.join(additional_parts.values()))
+        hash_parts.extend([timestamp, sid, origin])
+        sidhash = hashlib.sha1(' '.join(hash_parts).encode()).hexdigest()
+
+        parts = [timestamp, sidhash]
+        if additional_parts:
+            parts.append(''.join(additional_parts))
+
+        return f'{scheme} {"_".join(parts)}'
 
-    def _generate_sapisidhash_header(self, origin='https://www.youtube.com'):
-        time_now = round(time.time())
-        if self._SAPISID is None:
-            yt_cookies = self._get_cookies('https://www.youtube.com')
+    _SAPISID = _3PSAPISID = _1PSAPISID = None
+
+    def _load_sid_cookies(self):
+        yt_cookies = self._get_cookies('https://www.youtube.com')
+        yt_sapisid = try_call(lambda: yt_cookies['SAPISID'].value)
+        yt_3papisid = try_call(lambda: yt_cookies['__Secure-3PAPISID'].value)
+
+        if not self._SAPISID:
             # Sometimes SAPISID cookie isn't present but __Secure-3PAPISID is.
+            # YouTube also falls back to __Secure-3PAPISID if SAPISID is missing.
```

Comment:
> `# YouTube also falls back to __Secure-3PAPISID if SAPISID is missing.`

so this means that setting a SAPISID cookie with the 3P value was unnecessary, hence why we're not doing it anymore?

Reply:
correct